File: src/com/krishna/miniproject/Library.java
package com.krishna.miniproject;

import java.lang.reflect.Array;
import java.util.ArrayList;

public class Library {
    private ArrayList<Book> books;

    public Library(){
        books=new ArrayList<>();
    }

    public void addBook(Book book){
        books.add(book);
        System.out.println("Book added: "+book.getTitle());

    }
        public void removeBook(String bookId){
            for(int i=0;i<books.size();i++){
                if(books.get(i).getBookId().equals(bookId)){
                    System.out.println("book removed: "+books.get(i).getTitle());
                    books.remove(i);
                    return;
                }
            }
            System.out.println("book not found");
        }

        public void issueBook(String bookId, User user){
            for(int i=0;i<books.size();i++){
                if(books.get(i).getBookId().equals(bookId)){
                    if(books.get(i).isAvailable()){
                        books.get(i).issueBook();
                        System.out.println("Book issued to "+user.name +":" +books.get(i).getTitle());
                        return;
                    }
                    else{
                        System.out.println("Book is already issued");
                        return;
                    }
                }

            }
            System.out.println("Book not found");
        }

    public void returnBook(String bookId) {
        for(int i=0;i<books.size();i++) {
            if (books.get(i).getBookId().equals(bookId)) {
                if (!books.get(i).isAvailable()) {
                    books.get(i).returnBook();
                    System.out.println("Book returned: " + books.get(i).getTitle());
                    return;
                } else {
                    System.out.println("Book was not issued");
                    return;
                }

            }

        }
        System.out.println("Book not found");
    }

    public void showAvailableBooks() {
        boolean found=false;
        for(int i=0;i<books.size();i++){
            if(books.get(i).isAvailable()){
                found=true;
                System.out.println("Title: "+books.get(i).getTitle()+ ",Author:" + books.get(i).getAuthor()+ ",Id: "+books.get(i).getBookId());
            }
        }
        if(found==false){
            System.out.println("no books available");
        }
    }



}

File: Dart/src/krishna/miniproject/Library.dart
import 'Book.dart';
import 'User.dart';

class Library {
  List<Book> _books = [];

 Library(){
   
 }
 
 void addBook(Book book){
   _books.add(book);
   print("Book added: ${book.title}");
 }

 void removeBook(String bookId){
   for(int i=0;i<_books.length;i++){
     if(_books[i].bookId==bookId){
       print("Book removed: ${_books[i].title}");
       _books.removeAt(i);
       return;
     }
   }
   print("Book not found");
 }

 void issueBook(String bookId, User u){

   for(int i=0;i<_books.length;i++){
     if(_books[i].bookId==bookId){
       if(_books[i].isAvailable==true){
         print("Bood issued to ${u.userId} : ${_books[i].title}");
         _books[i].issueBook();
         return;
       }
       else{
         print("Book already issued");
         return;
       }
     }
   }
   print("Book not found");
 }

 void returnBook(String bookId){
   
   for(int i=0;i<_books.length;i++){
     if(_books[i].bookId==bookId){
       if(_books[i].isAvailable==false){
         print("Book returned: ${_books[i].title}");
         _books[i].returnBook();
         return;
       }
       else{
         print("Book was not issued");
         return;
       }
     }
   }
   print("Book not found");
 }

  void showAvailableBooks() {
   bool found=false;
    for(int i=0;i<_books.length;i++){
      if(_books[i].isAvailable){
        found=true;
        print("Title: ${_books[i].title}, Author: ${_books[i].author}, ID: ${_books[i].bookId}");
      }
    }
    if(found==false){
      print("No Books Available");
    }
  }


}
